### PR TITLE
Task/2015 more tests and cleanup

### PR DESCRIPTION
**Pull request by kzangeli**
Possible the last PR of this feature.
All traces gone and a few more tests added.
Also, a few broken functests about 'contextBroker usage' fixed.

### CODE REVIEW

File: test/functionalTest/cases/2015_notification_templates/notification_templates_cache_refresh.test [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription with template
+# 02. Sleep 2.1 seconds to assure the sub-cache is refreshed from mongo
+# 03. Create entity to trigger the subscription
+# 04. Dump accumulator to see the notification
+#
```
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_cache_refresh.test]: Step 5 is missing in the preamble.

File: test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED [hunk 2]
```diff
@@ -1,42 +0,0 @@
-# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED]: I understand this .test has been covered by other .test in previous PRs. Is that correct?

File: test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED [hunk 3]
```diff
@@ -1,44 +0,0 @@
-# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED]: I understand this .test has been covered by other .test in previous PRs. Is that correct?

File: test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED [hunk 4]
```diff
@@ -1,46 +0,0 @@
...
---NAME--
-Notification Templates, getting the cache from mongo at startup
-
---SHELL-INIT--
-dbInit CB
-brokerStart CB
-accumulatorStart --pretty-print
-
---SHELL--
-
```
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED]: I understand this .test has **not** been covered by other .test in previous PRs, but I understand you have a good reason for not using it at the end. Is that correct?

File: test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED [hunk 5]
```diff
@@ -1,40 +0,0 @@
-# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED]: I understand this .test has been covered by other .test in previous PRs. Is that correct?

File: CHANGES_NEXT_RELEASE [hunk 6]
```diff
@@ -10,3 +10,4 @@
 - Fix: -noCache now avoids populating cache on create/update subscription operations (#2147)
 - Deprecated: /ngsi10 and /ngsi9 as URL path prefixes
 - Fix: potential bug in internal logic at NGSIv2 GET subscription operations, possibly provoking crashes
+- Add: NGSIv2 notification templates (Issue #2015)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: At the end, taking into account how documentation and API artifacts (e.g. httpCustom) has evolved,I think that it is better to talk about "custom notifications", i.e.

"Add: NGSIv2 custom notifications (Issue #2015)"

File: src/lib/cache/subCache.cpp [hunk 7]
```diff
@@ -682,7 +682,6 @@ void subCacheItemInsert
   const std::string&            georel
 )
 {
-  LM_W(("KZ: inserting sub in cache. extended == %s", httpInfo.extended? "true" : "false"));
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: Just to confirm... code changes in this PR are only about removing KZ traces?

File: test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED (hunk 5 above)
Comment (kzangeli) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED]: Yes, see notification_templates_error_handling.test (steps 07-11)

File: test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED (hunk 4 above)
Comment (kzangeli) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED]: Yes, there is already a test that populates the sub-cache, then waits for the sub-cache to be refreshed and after that creates an entity to trigger notifications.
Manually I have verified that the sub.cache is refreshed (as you know, refreshing the cache is throwing it away and then repopulating it from mongo - almost identical to what wanted to do in this test).
See notification_templates_cache_refresh.test

File: test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED (hunk 3 above)
Comment (kzangeli) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED]: Yes, a textual payload is part of notification_templates_with_decoded_chars_in_notifications.test, step 01:

     "payload": "A string with double-quotes: %22${A1}%22"

File: test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED]: Yes, GET /v2/subscriptions is used in notification_templates_cache_refresh.test, step 05 (missing in TOC ;-))
Also used in notification_templates_with_decoded_chars_in_notifications.test

File: src/lib/cache/subCache.cpp (hunk 7 above)
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Yeah, just traces

File: test/functionalTest/cases/2015_notification_templates/notification_templates_cache_refresh.test (hunk 1 above)
Comment (kzangeli) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_cache_refresh.test]: Step 05 was a last-minute addition, nice catch!

Fixed in 9572f41

File: CHANGES_NEXT_RELEASE (hunk 6 above)
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: Fixed in 0fca14a

File: test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_get_subscription.test.DISABLED]: NTC

File: test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED (hunk 3 above)
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_payload_as_text.test.DISABLED]: NTC

File: test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED (hunk 4 above)
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_reread_cache_from_mongo.test.DISABLED]: NTC

File: test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED (hunk 5 above)
Comment (fgalan) [inline test/functionalTest/cases/2015_notification_templates/notification_templates_with_invalid_chars.test.DISABLED]: NTC

File: src/lib/cache/subCache.cpp (hunk 7 above)
Comment (fgalan) [inline src/lib/cache/subCache.cpp]: NTC

Let's do a careful look to next feature/2015_... -> develop PR in order to check that no new LM_W remains.

**Comment by fgalan** [comment]
LGTM

File: src/lib/cache/subCache.cpp (hunk 7 above)
Comment (kzangeli) [inline src/lib/cache/subCache.cpp]: Actually, that's what I did for this PR. I just might have removed a few originating from develop.
No 'bad' LM_W remaining. That said, always good to check again. Doesn't cost more than a few minutes.

**Pull request merged**